hax/hax/motr/__init__.py

```diff
@@ -405,17 +429,48 @@ def notify_node_status(self,
 
         node_fid = self.consul_util.get_node_fid(node)
         encl_fid = self.consul_util.get_node_encl_fid(node)
-        ctrl_fid = self.consul_util.get_node_ctrl_fid(node)
-        LOG.debug('node_fid: %s encl_fid: %s ctrl_fid: %s with state: %s',
-                  node_fid, encl_fid, ctrl_fid, new_state)
+        ctrl_fids = self.consul_util.get_node_ctrl_fids(node)
+        LOG.debug('node_fid: %s encl_fid: %s ctrl_fids: %s with state: %s',
+                  node_fid, encl_fid, ctrl_fids, new_state)
 
         notes = []
-        if node_fid and encl_fid and ctrl_fid:
+        if node_fid and encl_fid:
             notes = [HaNoteStruct(no_id=x.to_c(), no_state=new_state)
-                     for x in [node_fid, encl_fid, ctrl_fid]]
+                     for x in [node_fid, encl_fid]]
+
+        if int(len(ctrl_fids)) != 0:
+            for x in ctrl_fids:
+                notes.append(HaNoteStruct(no_id=x.to_c(), no_state=new_state))
 
         return notes
 
+    def notify_ctrl_status(self,
```

Comment:
The word "notify" is misleading. It notifies nobody, but just creates the HaNoteStruct if there is something to report about. Please think of a better name.

Reply:
Done